client/my-sites/plans-features-main/hooks/use-is-plan-upsell-enabled-on-free-domain.ts

```diff
@@ -0,0 +1,44 @@
+import { useExperiment } from 'calypso/lib/explat';
+import { useSelector } from 'calypso/state';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import type { DataResponse } from 'calypso/my-sites/plan-features-2023-grid/types';
+
+export const useIsPlanUpsellEnabledOnFreeDomain = (
+	flowName?: string | null,
+	freeSubDomainName?: string
+): DataResponse< boolean > => {
+	const userLocale = useSelector( getCurrentUserLocale );
```

Comment:
Overall the implementation looks a bit too terse. We can simpify it a bit by deriving the correct experiment name based on `flowName`. e.g.

```
const experimentName = flowName === 'onboarding' ?  'calypso_gf_signup_onboarding_free_free_dont_miss_out_modal' : 'calypso_gf_signup_onboarding_pm_free_free_dont_miss_out_modal' ;
```

Then the following `useExperiment()` call can be unified:

```
const [ isLoadingPaidMediaAssignment, experimentAssignment ] = useExperiment( experimentName, isEligible: ...
```

Reply:
> Overall the implementation looks a bit too terse.

I decided not to use your EXPERIMENT infrastructure for now since we are short on time.
And I am planning to refactor these approaches as soon as we ship the experiment.
So eventually I envision that there will be just 1 component <PlanUpsellModal /> And each case based component type will be routed inside this component.
And as you explained eventually we will consolidate the experiment assignment hooks as well in a meaningful way.





[12:26](https://a8c.slack.com/archives/D016ZDSM9E2/p1690950413064079)
So eventually I envision that there will be just 1 component <PlanUpsellModal /> And each case based component type will be routed inside this component.
And as you explained eventually we will consolidate the experiment assignment hooks as well in a meaningful way. (edited)

Reply:
I have gone ahead and done this change. Sorry I misunderstood your explanation above.